src/dynamic_boundary_conditions/hirds_depth_data_to_db.py

```diff
@@ -5,113 +5,144 @@
 @author: pkh35
 """
 
+import re
 import pandas as pd
-import geopandas
-import numpy as np
+import geopandas as gpd
 import sqlalchemy
+import pathlib
+import logging
 from shapely.geometry import Polygon
 from src.dynamic_boundary_conditions import rain_depth_data_from_hirds
+from src.digitaltwin import setup_environment
+from src.dynamic_boundary_conditions import hyetograph
 
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
 
-def check_table_exists(engine):
-    """Check if the hirds_rain_depth table exists in the database."""
+formatter = logging.Formatter("%(levelname)s:%(asctime)s:%(name)s:%(message)s")
+stream_handler = logging.StreamHandler()
+stream_handler.setFormatter(formatter)
+
+log.addHandler(stream_handler)
+
+
+def check_table_exists(engine, db_table_name: str) -> bool:
+    """Check if table exists in the database."""
     insp = sqlalchemy.inspect(engine)
-    return insp.has_table("hirds_rain_depth", schema="public")
-
-
-def get_sites_in_catchment(catchment_area: Polygon, engine) -> list:
-    """Get gauges within the catchment area from the 'hirds_gauges' table in
-    the database."""
-    query = f"""select site_id from public.hirds_gauges f
-        where ST_Intersects(f.geometry, ST_GeomFromText('{catchment_area}', 4326))"""
-    gauges_in_catchment = pd.read_sql_query(query, engine)
-    return gauges_in_catchment["site_id"].tolist()
-
-
-def get_sites_not_in_db(engine, sites_in_catchment):
-    """To only get the data for the sites for which data are not avialble in
-    the database."""
-    query = "SELECT DISTINCT site_id FROM hirds_rain_depth;"
-    gauges = engine.execute(query)
-    sites = gauges.fetchall()
-    sites = list(sites)
-    sites_in_db = []
-    for i in range(len(sites)):
-        sites_in_db.append(sites[i][0])
-    # yields the elements in `sites_in_catchment` that are NOT in `sites_in_db`
-    sites = np.setdiff1d(sites_in_catchment, sites_in_db)
-    return sites
-
-
-def get_data_from_csv(filename: str, site_id: str, rcp, time_period, n: int):
-    """Read data of the diffrent time period and return a dataframe."""
-    hirds_data = pd.read_csv(
-        filename, skiprows=n, nrows=12, index_col=None, quotechar='"'
-    )
-    hirds_data["site_id"] = site_id
-    hirds_data["rcp"] = rcp
-    hirds_data["time_period"] = time_period
-    hirds_data.columns = hirds_data.columns.str.lower()
-    return hirds_data
-
-
-def add_hirds_depth_data_to_db(path: str, site_id: str, engine):
-    """Store each site's depth data in the database. Each csv file contains
-    data for historical rainfall depth, and for various rcp and time period.
-    To view the csv file, go to : https://hirds.niwa.co.nz/, select a site and
-    generate a report, there are rainfall depths for diffrent RCP Scenarios.
+    return insp.has_table(db_table_name, schema="public")
+
+
+def get_sites_id_in_catchment(engine, catchment_polygon: Polygon) -> list:
```

Comment:
When using list as a type you can also use it like this for even more type info:
```python
from typing import List

def list_of_numbers_0_to_n(n: int) -> List[int]:
   result = []
   for i in range(n):
       result.append(i)
   return result
```
If this is easy to add this stronger type annotation then add it here, or if not then reply to the comment

Reply:
Done :)